RelVal/o2dpg_release_validation.py

```diff
@@ -235,14 +235,23 @@ def file_sizes(dirs, threshold):
     return collect_dict
 
 
-def plot_pie_charts(summary, out_dir, title):
+def plot_pie_charts(summary, out_dir, title, include_patterns=None):
 
     print("==> Plot pie charts <==")
 
     test_n_hist_map = {}
 
     # need to re-arrange the JSON structure abit for per-test result pie charts
     for histo_name, tests in summary.items():
+        # check if histo_name is in include patterns
+        if include_patterns:
+            include_this = False
+            for ip in include_patterns:
+                if re.search(ip,histo_name):
+                    include_this = True
+                    break
+            if not include_this:
+                continue
```

Comment:
This appears now in at least 4 places. Could you put that into a small helper function?